File: ruby/HackerRank/the-birthday-bar.rb
len = gets().chomp().to_i()
count = 0
chocs = gets().chomp().split().map(&:to_i)
n_sums, n_parts = gets.chomp().split().map(&:to_i)


0.upto(len-1) do |i|
    if chocs[i...i+n_parts].sum() == n_sums
        count += 1
    end
end


puts(count)

File: Competive Programming/Python/HackerRank/the-birthday-bar.py
cases = int(input())
count = 0
chocs = list(map(int, input().split()))
n_sum, n_parts = map(int, input().split())


for i in range(cases):
    if sum(chocs[i:i+n_parts]) == n_sum:
        count += 1


print(count)
